### PR TITLE
(feat) precise algo type in search log events

### PR DESCRIPTION
**Pull request by rmelisson**
add algo type to select_result and use "both" algo when fulltext+sem are matching

### CODE REVIEW

File: packages/code-du-travail-frontend/src/search/SearchResults/index.js [hunk 1]
```diff
@@ -13,11 +13,16 @@ const SearchResults = ({
 }) => {
   const router = useRouter();
   useEffect(() => {
-    const toSlug = ({ source, slug }) => `${source}/${slug}`;
+    const toLogCandidate = ({ source, slug, algo }) => {
+      return {
+        slug: `${source}/${slug}`,
```
Comment (lionelB) [inline packages/code-du-travail-frontend/src/search/SearchResults/index.js]: should we use getRouteBySource here (or remove it from upper code ?)

Comment (rmelisson) [reply packages/code-du-travail-frontend/src/search/SearchResults/index.js]: Ah yes good find, for now the urls we send are different, I was making the match in the logs analysis, but that should be done properly here. 
I guess this is the recommended way to format it ?
```
const trackedUrl = 
   source === SOURCES.EXTERNALS ? url : `/${getRouteBySource(source)}/${slug}`;
```

Comment (lionelB) [reply packages/code-du-travail-frontend/src/search/SearchResults/index.js]: yep !

Comment (rmelisson) [reply packages/code-du-travail-frontend/src/search/SearchResults/index.js]: done

**Review by lionelB — APPROVED** [review]

**Pull request merged**